File: scala/nth-prime/src/test/scala/NthPrimeTest.scala
import org.scalatest.{Matchers, FunSuite}

/** @version 2.1.0 */
class NthPrimeTest extends FunSuite with Matchers {
  
  test("first prime") {
    NthPrime.prime(1) should be (Some(2))
  }

  test("second prime") {
    NthPrime.prime(2) should be (Some(3))
  }

  test("sixth prime") {
    NthPrime.prime(6) should be (Some(13))
  }

  test("big prime") {
    NthPrime.prime(10001) should be (Some(104743))
  }

  test("there is no zeroth prime") {
    NthPrime.prime(0) should be (None)
  }
}

File: typescript/nth-prime/nth-prime.test.ts
import { nth } from "./nth-prime";

describe("Prime", () => {
  it("first", () => {
    expect(nth(1)).toEqual(2);
  });

  it("second", () => {
    expect(nth(2)).toEqual(3);
  });

  it("sixth", () => {
    expect(nth(6)).toEqual(13);
  });

  it("big prime", () => {
    expect(nth(10001)).toEqual(104743);
  });

  it("weird case", () => {
    expect(() => nth(0)).toThrowError("Prime is not possible");
  });
});
